src/clj/commiteth/db/core.clj

```diff
@@ -92,3 +93,30 @@
 
 (defn update! [& args]
   (apply jdbc/update! *db* args))
+
+(defn convert-keys-to-lisp-case [res]
+  (->> res
+       (map #(vector (keyword (str/replace (name (first %1)) "_" "-")) 
+                     (second %1)))
+       (into {})))
+
+(defn result-one-sql->lisp
+  [this result options]
+  (convert-keys-to-lisp-case (first result)))
+
+(defn result-many-sql->lisp
+  [this result options]
+  (map convert-keys-to-lisp-case result))
+
+(defmethod hugsql.core/hugsql-result-fn :1 [sym]
+  'commiteth.db.core/result-one-sql->lisp)
+
+(defmethod hugsql.core/hugsql-result-fn :one [sym]
+  'commiteth.db.core/result-one-sql->lisp)
+
+(defmethod hugsql.core/hugsql-result-fn :* [sym]
+  'commiteth.db.core/result-many-sql->lisp)
+
+(defmethod hugsql.core/hugsql-result-fn :many [sym]
```

Comment:
These seem heavy duty, good stuff! 👏  Is the correct reading of this that they are extending a multimethod defined within the hugsql code? When are they called exactly?

Reply:
This was taken from https://github.com/layerware/hugsql/issues/21. These methods override built-in ones.

Reply:
cool, thanks for the link. really cool to see multimethods used in the wild like this.